src/border.jl

```diff
@@ -256,11 +574,108 @@ struct Fill{T,N} <: AbstractBorder
     Fill{T,N}(value::T, lo::Dims{N}, hi::Dims{N}) where {T,N} = new{T,N}(value, lo, hi)
 end
 
+"""
+```julia
+Fill(value::T)
+```
+
+Construct an instance of [`Fill`](@ref) designating a `value` and zero padding (i.e. no padding).
+
+---
+
+"""
 Fill(value::T) where {T} = Fill{T,0}(value)
+
+"""
+```julia
+    Fill(value::T, lo::Dims{N}, hi::Dims{N})
+```
+Construct an instance of [`Fill`](@ref) designating a `value`  and tuples
+`lo` and `hi` which stipulate the number of row and columns which will be predended
+and appended to the image.
+
+The tuple `lo` specifies how many row and columns to prepend the
+image with—the padding for the top and left border. The tuple `hi` specifies
+how many pixels to append the image with—the padding for the bottom and right
+border.
+
+#### Usage illustration
+Use `Fill(5,(2,2),(2,2))` to specify a padding of two pixels for the top,
+bottom, left and right edge with the value five.
+
+
+Use `Fill(zero(eltype(img))(1,2),(3,4))` to specify a padding of one, two, three
+and four pixels for the top, left, bottom and right edge respectively using a
+value of zero with the same type as `img`.
+
+
+---
+
+"""
 Fill(value::T, lo::Dims{N}, hi::Dims{N}) where {T,N} = Fill{T,N}(value, lo, hi)
+
+"""
+```julia
+    Fill(value::T, both::Dims{N})
+```
+Construct an instance of [`Fill`](@ref) designating a `value`  and a tuple
+`both` which stipulate the number of row and columns which will be prepended
+and appended to the image.
+
+#### Usage illustration
+Use `Fill(0,(5,10))` to stipulate a padding of five pixels for the top and left
+edge, and a padding of ten pixels for the bottom and right edge with a value of
+zero.
+
+
+---
+
+"""
 Fill(value::T, both::Dims{N}) where {T,N} = Fill{T,N}(value, both, both)
+
+"""
+```julia
+    Fill(value, lo::AbstractVector, hi::AbstractVector)
```

Comment:
Here you say `lo` is an `AbstractVector` but the text talks about a tuple. The code accepts either. Perhaps just don't put a type on these arguments? Then you only need to document one of them. Here too you could alternatively use the style

```julia
    Fill(value::T, lo::Dims{N}, hi::Dims{N})
    Fill(value::T, lo::AbstractVector, hi::AbstractVector)

Construct an instance...
```

Reply:
Done